dm/ctl/master/query_status.go

```diff
@@ -60,5 +75,75 @@ func queryStatusFunc(cmd *cobra.Command, _ []string) {
 		return
 	}
 
-	common.PrettyPrintResponse(resp)
+	if resp.Result && taskName == "" && len(workers) == 0 {
+		result := wrapTaskResult(resp)
+		common.PrettyPrintInterface(result)
+	} else {
+		common.PrettyPrintResponse(resp)
+	}
+}
+
+// errorOccurred checks ProcessResult and return true if some error occurred
+func errorOccurred(result *pb.ProcessResult) bool {
+	return result != nil && len(result.Errors) > 0
+}
+
+// getRelayStage returns current relay stage (including stageError)
+func getRelayStage(relayStatus *pb.RelayStatus) string {
+	if errorOccurred(relayStatus.Result) {
+		return stageError
+	}
+	return relayStatus.Stage.String()
+}
+
+// wrapTaskResult picks task info and generate tasks' status and relative workers
+func wrapTaskResult(resp *pb.QueryStatusListResponse) *taskResult {
+	taskStatusMap := make(map[string]string, len(resp.Workers))
+	taskCorrespondingWorkers := make(map[string][]string, len(resp.Workers))
+	for _, worker := range resp.Workers {
+		relayStatus := worker.RelayStatus
+		for _, subTask := range worker.SubTaskStatus {
+			subTaskName := subTask.Name
+			subTaskStage := subTask.Stage
+
+			taskCorrespondingWorkers[subTaskName] = append(taskCorrespondingWorkers[subTaskName], worker.Worker)
+			taskStage := taskStatusMap[subTaskName]
+			// the status of a task is decided by its subtasks, the rule is listed as follows:
+			// |                     Subtasks' status                       |                Task's status                 |
+			// | :--------------------------------------------------------: | :------------------------------------------: |
+			// |           Any Paused and len(result.errors) > 0            |    Error - Some error occurred in subtask    |
+			// | Any Running and unit is "Sync" and relay is Paused/Stopped | Error - Relay status is Error/Paused/Stopped |
+			// |              Any Paused but without error                  |                    Paused                    |
+			// |                        All New                             |                     New                      |
+			// |                      All Finished                          |                   Finished                   |
+			// |                      All Stopped                           |                   Stopped                    |
```

Comment:
seems we can't get `New/Finished/Stopped` status from current logic. Shall we assign the value to taskStatusMap for each first loop index?

Reply:
`case taskStage == "", subTaskStage == pb.Stage_Paused` will be satisfied for each first loop index and the value of `subTaskStage` will be assigned to `taskStatusMap`.